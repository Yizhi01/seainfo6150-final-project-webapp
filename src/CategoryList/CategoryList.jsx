import React from "react";
import { Link } from "react-router-dom";
import styles from "./CategoryList.module.css";
import PropTypes from "prop-types";

const CategoryList = (props) => {
  let displayContent;

  if (props.categorylist.length) {
    displayContent = props.categorylist.map((ca) => (
      <div key={ca.id}>
        <Link to={"/category/" + ca.id +'/' + ca.name} className={styles.categoryItem}>
          <img
            className={styles.categoryImage}
            src={ca.image._url}
            alt={ca.name}
          />
          <p className={styles.categoryName}>{ca.name}</p>
        </Link>
      </div>
  
    ));
  } else {
    displayContent = <div>You have no data!</div>;
  }

  return (
    <div className={styles.categoryContainer}>
      <h1 className={styles.title}>Recipes</h1>
      <div className={styles.categoryGrid}>
        {displayContent}
      </div>
    </div>
  )
}

CategoryList.propTypes = {
  categorylist: PropTypes.array.isRequired
};

export default CategoryList;