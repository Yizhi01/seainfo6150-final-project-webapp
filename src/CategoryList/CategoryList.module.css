.categoryContainer {
  display: flex;
  flex-direction: column;
}

.title {
  margin: 30px 0;
  text-align: center;
  font: italic 40px 'Castoro', serif;
  color: #707070;
}

.categoryGrid {
  margin: 0 100px 30px;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 50px;
}

@media (max-width: 900px) {
  .categoryGrid {
    grid-template-columns: repeat(2, 1fr);
  }
}

@media (max-width: 600px) {
  .categoryGrid {
    grid-template-columns: 1fr;
  }
}
