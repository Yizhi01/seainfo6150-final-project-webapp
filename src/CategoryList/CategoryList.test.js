import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import CategoryList from "./CategoryList.jsx";

const testlist = [
  { 
    "id": "1", 
    "name": "breakfast",
    "image": {
      "_url": "https://source.unsplash.com/Wns3U-oVoLA/960x960",
      "_height": "960",
      "_width": "960"
    }
  }
];

it("renders the CategoryList component correctly", () => {
  const { container } = render(
    <Router>
      <CategoryList categorylist={testlist} />
    </Router>
  );
  expect(container).toMatchSnapshot();
});