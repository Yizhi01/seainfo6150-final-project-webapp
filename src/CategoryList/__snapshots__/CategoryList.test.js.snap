// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CategoryList component correctly 1`] = `
<div>
  <div
    class="categoryContainer"
  >
    <h1
      class="title"
    >
      Recipes
    </h1>
    <div
      class="categoryGrid"
    >
      <div>
        <a
          class="categoryItem"
          href="/category/1/breakfast"
        >
          <img
            alt="breakfast"
            class="categoryImage"
            src="https://source.unsplash.com/Wns3U-oVoLA/960x960"
          />
          <p
            class="categoryName"
          >
            breakfast
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
